### ISSUE TITLE
Something  wrong  when chose GRU  instead of LSTM  bug 

### ISSUE THREAD

**Issue by 978749951** [OpenNMT/OpenNMT-py#1069]
@smartkiwi @srush @clementfarabet when I run the command 
"CUDA_VISIBLE_DEVICES=3 python train.py -data data/bpe15k -save_model model/bpe15k rld_size 1 -gpu_ranks 0 -rnn_type LSTM"    the  program work fine.  The output is as follows:

[2018-11-23 16:41:44,837 INFO] Loading train dataset from data/bpe15k.train.0.pt, number of examples: 189557
[2018-11-23 16:41:49,278 INFO] Step 50/100000; acc:   4.03; ppl: 42610.88; xent: 10.66; lr: 1.00000; 10579/15225 tok/s;      4 sec
[2018-11-23 16:41:53,194 INFO] Step 100/100000; acc:   7.22; ppl: 3426.26; xent: 8.14; lr: 1.00000; 12657/17596 tok/s;      8 sec
[2018-11-23 16:41:57,054 INFO] Step 150/100000; acc:  13.26; ppl: 1103.89; xent: 7.01; lr: 1.00000; 11771/17214 tok/s;     12 sec
[2018-11-23 16:42:01,278 INFO] Step 200/100000; acc:  13.64; ppl: 693.16; xent: 6.54; lr: 1.00000; 12294/16742 tok/s;     16 sec
[2018-11-23 16:42:05,019 INFO] Step 250/100000; acc:  16.42; ppl: 375.91; xent: 5.93; lr: 1.00000; 12811/18007 tok/s;     20 sec

However, when change rnn_type to GRU, i got a puzzling output:

[2018-11-23 16:43:59,840 INFO] Loading train dataset from data/bpe15k.train.0.pt, number of examples: 189557
[2018-11-23 16:44:04,080 INFO] Step 50/100000; acc:   2.64; ppl: 32042891.97; xent: 17.28; lr: 1.00000; 10707/15405 tok/s;      4 sec
[2018-11-23 16:44:08,346 INFO] Step 100/100000; acc:   3.41; ppl: 339616176.40; xent: 19.64; lr: 1.00000; 12206/16921 tok/s;      9 sec
[2018-11-23 16:44:12,386 INFO] Step 150/100000; acc:   2.87; ppl: 163551401.84; xent: 18.91; lr: 1.00000; 12038/16945 tok/s;     13 sec

the value  of  acc  and ppl  indicate that there must something wrong.  I just clone the code from github and run it without any edition. Months ago, I used a older version "OpenNMT" and it work perfect.

**Comment by vince62s** [comment]
Can you please tell which older version was fine ?

**Comment by 978749951** [comment]
I download the older version OpenNMT on April 15th. However, the code has changed a lot.

**Comment by vince62s** [comment]
I can look into it, but can you test the releases from here: https://github.com/OpenNMT/OpenNMT-py/releases
starting from June upward, just to help identify where it may come from.
thanks

**Comment by 978749951** [comment]
I test the releases from [https://github.com/OpenNMT/OpenNMT-py/releases](url) starting from june upward.  
I find that all version (v0.1  v0.2  v0.2.1  v0.3.0  v0.4.0 v0.4.1 and v0.5.0) work fine with "-rnn_type LSTM", but none of them works with "-rnn_type GRU".

**Comment by guillaumekln** [comment]
GRU requires different optimization settings, typically a lower learning rate when using SGD.

**Comment by 978749951** [comment]
The problem was Solved!
@guillaumekln  thanks a lot. As the learning rate is default item, i even ignore  the significant effect of learning rate and optimizer.
@vince62s  I am sorry to bother you.  The code is no problem!
